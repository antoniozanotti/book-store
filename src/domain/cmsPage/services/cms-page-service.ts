import { CmsPageWithPagePrisma } from "../types/cms-prisma";
import prisma from "@/lib/db";

export async function getCmsPageBySlug(
  slug: string
): Promise<CmsPageWithPagePrisma> {
  return await prisma.cmsPage.findFirstOrThrow({
    where: {
      page: {
        slug: slug,
      },
    },
    include: {
      page: true,
    },
  });
}
