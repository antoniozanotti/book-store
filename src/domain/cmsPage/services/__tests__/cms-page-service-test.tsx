/**
 * @jest-environment node
 */
import { cmsPageWithPageMock } from "../../__tests__/cms-page-with-page-mock";
import { getCmsPageBySlug } from "../cms-page-service";

describe("CmsPageService", () => {
  it("should return cms page by slug", async () => {
    const cmsPage = await getCmsPageBySlug(cmsPageWithPageMock.page.slug);
    expect(cmsPage).toMatchSnapshot();
  });
});