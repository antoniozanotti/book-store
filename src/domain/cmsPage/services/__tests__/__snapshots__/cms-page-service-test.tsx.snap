// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CmsPageService should return cms page by slug 1`] = `
{
  "content": "Color acceptus usitas ultio basium ascisco supplanto tametsi adstringo volup. Comminor perspiciatis universe laboriosam vulnero maiores cibus acsi assentator adicio. Coadunatio pel baiulus titulus sordeo tamquam aperio verbera.
Pecco demoror timidus sub a dedico urbs articulus aiunt. Cognatus sordeo vergo cupio alias crebro theologus tunc error. Claustrum subseco ventus sordeo aliquam vulgo.
Velociter territo voluntarius cavus delectatio culpo beneficium universe contigo earum. Patrocinor curvo voluptate dolorem delego vir. Facere vitae vir concido.",
  "id": 13,
  "page": {
    "description": "Voluptate utrum veritatis bibo velociter deporto curia celebrer via sollicito. Tricesimus aspernatur vigor officiis. Usque stillicidium velum.",
    "id": 13,
    "slug": "about-us",
    "title": "About Us",
    "type": "cmsPage",
  },
  "pageId": 13,
  "title": "About Us",
}
`;
